src/clients/BundleDataClient/utils/FillUtils.ts

```diff
@@ -75,20 +83,24 @@ export async function verifyFillRepayment(
   _fill: FillWithBlock,
   destinationChainProvider: providers.Provider,
   matchedDeposit: DepositWithBlock,
-  possibleRepaymentChainIds: number[] = []
+  hubPoolClient: HubPoolClient
 ): Promise<FillWithBlock | undefined> {
   const fill = _.cloneDeep(_fill);
 
-  const repaymentChainId = getRepaymentChainId(fill, matchedDeposit);
-  const validEvmRepayment = isEvmRepaymentValid(fill, repaymentChainId, possibleRepaymentChainIds);
-
-  // Case 1: Repayment chain is EVM and repayment address is valid EVM address.
-  if (validEvmRepayment) {
+  // Slow fills don't result in repayments so they're always valid.
+  if (isSlowFill(fill)) {
     return fill;
   }
-  // Case 2: Repayment chain is EVM but repayment address is not a valid EVM address. Attempt to switch repayment
-  // address to msg.sender of relay transaction.
-  else if (chainIsEvm(repaymentChainId) && !isValidEvmAddress(fill.relayer)) {
+
+  let repaymentChainId = getRepaymentChainId(fill, matchedDeposit);
+
+  // If repayment chain doesn't have a Pool Rebalance Route for the input token, then change the repayment
+  // chain to the destination chain.
+  if (!isSlowFill(fill) && forceDestinationRepayment(repaymentChainId, matchedDeposit, hubPoolClient)) {
```

Comment:
We already return early for a slow fill so the initial check of `isSlowFill()` should be redundnat.

Reply:
ah yes good catch

Reply:
https://github.com/across-protocol/sdk/pull/863/commits/68b52fce4f3cb32350a288de1aa64e9cbd718bae